### PR TITLE
replay(feat): Adjust default Replay>Network Panel height & min/max sizes

### PR DESCRIPTION
**Pull request by ryan953**
Update the min/max drag size for the Replay>Network Panel

Now when you click to open a row, that row will be scrolled into view when the panel opens.
| Init | After Click |
| --- | --- |
| init | after click | 

- When dragging the panel larger we don't scroll the items behind. So the open item might not be visible.
- When opening a link that includes `?n_detail_row=23&n_detail_tab=details` in the url, the panel will open by default, but we won't auto-scroll the table to make the row visible. So the open item might not be visible.

When dragging to make the panel larger, the new max-size will ensure that at least one row is visible at all times. It doesn't have to be the row that is currently open though!

SCR-20230428-jfub

This means that you can expand the panel to be very large, and then click the "eye" icon and see that row up at the top.

If you resize the window (like make it shorter) then the max size is not respected -> the panel can cover the table headers and all table rows.

### CODE REVIEW

File: static/app/views/replays/detail/network/index.tsx [hunk 1]
```diff
@@ -69,22 +70,33 @@ function NetworkList({networkSpans, startTimestampMs}: Props) {
...
     });
 
-  const initialRequestDetailsHeight = useMemo(
-    () => Math.max(150, window.innerHeight * 0.25),
-    []
-  );
   const {size: containerSize, ...resizableDrawerProps} = useResizableDrawer({
     direction: 'up',
-    initialSize: initialRequestDetailsHeight,
+    initialSize: Math.max(150, window.innerHeight * 0.4), // cannot depend on containerRef :(
```
Comment (ryan953) [inline static/app/views/replays/detail/network/index.tsx:75]: larger size from before

I tried making this relative to `containerRef` but container ref starts as undefined, then `initialSize` gets cached by the hook and doesn't update :(

I briefly looked at what it takes to change that, but i don't think it's worth the complexity.

Comment (billyvg) [reply static/app/views/replays/detail/network/index.tsx:75]: Would be good to add why you can't depend on containerRef in the comment

File: static/app/views/replays/detail/network/index.tsx [hunk 2]
```diff
@@ -69,22 +70,33 @@ function NetworkList({networkSpans, startTimestampMs}: Props) {
...
     'n_detail_row',
     ''
   );
-  const detailItemIndex = getDetailRow();
-  const splitSize = networkSpans && detailItemIndex ? containerSize : undefined;
+  const detailDataIndex = getDetailRow();
+  const detailRowIndex = Number(detailDataIndex) + 1;
+
+  const maxContainerHeight =
+    (containerRef.current?.clientHeight || window.innerHeight) - 90;
```
Comment (billyvg) [inline static/app/views/replays/detail/network/index.tsx:87]: What does the `90` represent here?

**Review by billyvg — COMMENTED** [review]
The icon ends up making the row taller than the others: 
image

File: static/app/views/replays/detail/network/index.tsx (hunk 2 above)
Comment (ryan953) [reply static/app/views/replays/detail/network/index.tsx:87]: made a const and renamed to `RESIZEABLE_HANDLE_HEIGHT`

**Comment by ryan953** [comment]
> The icon ends up making the row taller than the others:

ya, also got the feedback that the icon is not intuitive "whats this do". So we'll leave it alone for this PR but come back and do something different with it.

**Review by billyvg — APPROVED** [review]

**Pull request merged**